agent/cache/cache.go

```diff
@@ -452,7 +458,7 @@ func (c *Cache) fetch(key string, r getOptions, allowNew bool, attempt uint, ign
 	// If we don't have an entry, then create it. The entry must be marked
 	// as invalid so that it isn't returned as a valid value for a zero index.
 	if !ok {
-		entry = cacheEntry{Valid: false, Waiter: make(chan struct{})}
+		entry = cacheEntry{Valid: false, Waiter: make(chan struct{}), RateLimiter: rate.NewLimiter(c.options.EntryFetchRateLimit, c.options.EntryFetchMaxBurst)}
```

Comment:
minor: now that this is really long it might read better with newlines for each field:

```suggestion
		entry = cacheEntry{
			Valid: false,
			Waiter: make(chan struct{}),
			RateLimiter: rate.NewLimiter(c.options.EntryFetchRateLimit, 1),
		}
```

Reply:
Fixed in https://github.com/hashicorp/consul/pull/8226/commits/1e0b1d2886f2108d18f4385d7ed1625ae6a946ec